packages/@aws-cdk/aws-redshift/lib/parameter-group.ts

```diff
@@ -62,17 +113,37 @@ export class ClusterParameterGroup extends ClusterParameterGroupBase {
    */
   public readonly clusterParameterGroupName: string;
 
+  /**
+   * The parameters in the parameter group
+  */
+  readonly parameters: { [name: string]: string };
+
   constructor(scope: Construct, id: string, props: ClusterParameterGroupProps) {
     super(scope, id);
-
+    this.parameters = props.parameters;
     const resource = new CfnClusterParameterGroup(this, 'Resource', {
       description: props.description || 'Cluster parameter group for family redshift-1.0',
       parameterGroupFamily: 'redshift-1.0',
-      parameters: Object.entries(props.parameters).map(([name, value]) => {
-        return { parameterName: name, parameterValue: value };
-      }),
+      parameters: Lazy.any({ produce: () => this.parseParameters() }),
     });
 
     this.clusterParameterGroupName = resource.ref;
   }
+  private parseParameters(): any {
+    return Object.entries(this.parameters).map(([name, value]) => {
+      return { parameterName: name, parameterValue: value };
+    });
+  }
+
+  public addParameter(name: string, value: string): AddParameterResult {
+    const existingValue = Object.entries(this.parameters).find(([key, _]) => key === name)?.[1];
+    if (existingValue === undefined) {
+      this.parameters[name] = value;
+      return { parameterAddedResult: AddParameterResultStatus.SUCCESS };
+    } else if (existingValue === value) {
+      return { parameterAddedResult: AddParameterResultStatus.SAME_VALUE_FAILURE };
+    } else {
+      return { parameterAddedResult: AddParameterResultStatus.CONFLICTING_VALUE_FAILURE };
```

Comment:
Throw error messages instead of returning. You're not getting anything out of any of these return values.

Reply:
Touched upon it [here](https://github.com/aws/aws-cdk/pull/20944/files#r937243248). Probably better to remove the `parameterAddedResult` entirely and just throw the error in the case of a mismatched parameter value